tests/ui/union.rs

```diff
@@ -0,0 +1,9 @@
+#[allow(unused_imports)]
+use core::marker::PhantomData;
+
+#[derive_where::derive_where(Clone)]
+union Test<T> {
+    field: PhantomData<T>,
+}
```

Comment:
Should we create an issue for unions? To document we could implement them if needed.

Reply:
I briefly looked into this yesterday, just to make sure why we can't implement them. The issue is that we can't handle unions, because unlike an `enum`, we have no way to check which current variant is active. So really the only trait we could derive for unions is `Copy`, and therefore a special implementation of `Clone`, and `Hash` (I think?).

On that note, standard `derive` only support `Copy`.
So I guess yes, we should create an issue!

Reply:
See #18.